tests/client/disco/TestAddonManager.js

```diff
@@ -49,31 +52,46 @@ describe('addonManager', () => {
   });
 
   describe('install()', () => {
-    it('should call mozAddonManager.createInstall() with url', () => {
-      addonManager.install(fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager });
-      assert.ok(fakeMozAddonManager.createInstall.calledWith({ url: fakeInstallUrl }));
-    });
-
-    it('should call installObj.addEventListener to setup events', () => addonManager.install(
-      fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+    it(
+      'should call mozAddonManager.createInstall() with url',
+      () => addonManager.install(
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
         .then(() => {
-          assert.equal(fakeInstallObj.addEventListener.callCount, installEventList.length);
+          assert.ok(fakeMozAddonManager.createInstall.calledWith({ url: fakeInstallUrl }));
         }));
 
+    it(
+      'should call installObj.addEventListener to setup events',
+      () => addonManager.install(
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+          .then(() => {
+            // It registers an extra onInstallFailed and onInstallEnded listener.
+            assert.equal(fakeInstallObj.addEventListener.callCount, installEventList.length + 2);
+          }));
+
     it('should call installObj.install()', () => addonManager.install(
       fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
         .then(() => {
           assert.ok(fakeInstallObj.install.called);
         }));
 
+    it('rejects if the install fails', () => {
+      fakeInstallObj.install = sinon.spy(function install() {
+        this.onInstallFailedListener();
+      });
+      return addonManager.install(
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+        .then(
+          () => assert.ok(false, 'expected promise to reject'),
```

Comment:
You can use `unexpectedSuccess` https://github.com/mozilla/addons-frontend/blob/master/tests/client/helpers.js#L32 for this.